packages/tpetra/core/ext/TpetraExt_MatrixMatrix_HIP.hpp

```diff
@@ -0,0 +1,781 @@
+            "(c_status = " << c_status[Cij] << " of [" << OLD_ip << "," << CSR_ip << "))");
+
+          Cvals[c_status[Cij]] += Aval * Ivals[j];
+        }
+      }
+    }
+  }
+
+  C.fillComplete(C.getDomainMap(), C.getRangeMap());
+}
+
+/*********************************************************************************************************/
+template<class Scalar,
+         class LocalOrdinal,
+         class GlobalOrdinal,
+         class LocalOrdinalViewType>
+void KernelWrappers2<Scalar,LocalOrdinal,GlobalOrdinal,Kokkos::Compat::KokkosHIPWrapperNode,LocalOrdinalViewType>::jacobi_A_B_newmatrix_kernel_wrapper(Scalar omega,
+                                                                                               const Vector<Scalar,LocalOrdinal,GlobalOrdinal,Kokkos::Compat::KokkosHIPWrapperNode> & Dinv,
+                                                                                               CrsMatrixStruct<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosHIPWrapperNode>& Aview,
+                                                                                               CrsMatrixStruct<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosHIPWrapperNode>& Bview,
+                                                                                               const LocalOrdinalViewType & Acol2Brow,
+                                                                                               const LocalOrdinalViewType & Acol2Irow,
+                                                                                               const LocalOrdinalViewType & Bcol2Ccol,
+                                                                                               const LocalOrdinalViewType & Icol2Ccol,
+                                                                                               CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosHIPWrapperNode>& C,
+                                                                                               Teuchos::RCP<const Import<LocalOrdinal,GlobalOrdinal,Kokkos::Compat::KokkosHIPWrapperNode> > Cimport,
+                                                                                               const std::string& label,
+                                                                                               const Teuchos::RCP<Teuchos::ParameterList>& params) {
+
+#ifdef HAVE_TPETRA_MMM_TIMINGS
+    std::string prefix_mmm = std::string("TpetraExt ") + label + std::string(": ");
+    using Teuchos::TimeMonitor;
+    Teuchos::RCP<TimeMonitor> MM;
+#endif
+
+  // Node-specific code
+  using Teuchos::RCP;
+
+  // Options
+  //int team_work_size = 16;  // Defaults to 16 as per Deveci 12/7/16 - csiefer // unreferenced
+  std::string myalg("KK");
+  if(!params.is_null()) {
+    if(params->isParameter("cuda: jacobi algorithm"))
+      myalg = params->get("cuda: jacobi algorithm",myalg);
```

Comment:
```suggestion
    if(params->isParameter("hip: jacobi algorithm"))
      myalg = params->get("hip: jacobi algorithm",myalg);
```

Reply:
@masterleinad Good suggestion, I changed all these lines in one commit.